packages/ember-routing/lib/location/api.js

```diff
@@ -99,20 +105,43 @@ export default {
     return implementationClass.create(...arguments);
   },
 
-  implementations: {},
-  _location: location,
-
-  /**
-    Returns the current `location.hash` by parsing location.href since browsers
-    inconsistently URL-decode `location.hash`.
-
-    https://bugzilla.mozilla.org/show_bug.cgi?id=483304
+  _implementations: {},
+};
 
-    @private
-    @method getHash
-    @since 1.4.0
-  */
-  _getHash() {
-    return getHash(this.location);
+if (DEBUG && HAS_NATIVE_PROXY) {
+  /* globals Proxy Reflect */
+  Location._implementations = new Proxy(
```

Comment:
Hmm, this strikes me as odd. Did we previously support `Location._implementations`?

Reply:
I just put depreciation for `Location.implementations = { location: MyLocation }` case, when assigned in one go